### ISSUE TITLE
Monitor task with 0.8 openshift-pipelines operator picks up wrong service account name

### ISSUE THREAD

**Issue by a-roberts** [tektoncd/experimental#388]
With the operator (canary channel):

```
adams-mbp:newappsody aroberts$ k get taskrun monitor-taskrun-22qbz -o yaml | grep service
  serviceAccount: tekton-webhooks-extension
  serviceAccountName: pipeline
```
I created my webhook with the SA as tekton-webhooks-extension, pipeline is a built-in SA which doesn't have the permissions I need so my monitor task gets an RBAC error

Without

```
adams-mbp:newappsody aroberts$ k get pod monitor-taskrun-q4mmr-pod-64d031 -o yaml | grep serviceAccount
  serviceAccount: tekton-webhooks-extension
  serviceAccountName: tekton-webhooks-extension
```

@dibbles spotted https://github.com/openshift/tektoncd-pipeline-operator/blob/master/pkg/flag/flag.go which is potentially a clue (notice the SA name defaulting to `pipeline`)

**Comment by a-roberts** [comment]
@vdemeester any thoughts please? 👼

**Comment by mnuttall** [comment]
I'm hoping there's a flag on the operator that can be used to set a default serviceAccount. @nikhil-thomas or @sthaha may also be able to advise?

**Comment by sthaha** [comment]
@a-roberts @mnuttall I am closing this in favour of https://github.com/openshift/tektoncd-pipeline-operator/issues/139 (copied the issue to the acutal repo)

**Comment by a-roberts** [comment]
Thanks @sthaha for taking a look, reopened due to the comment Vincent's made here:

> This will be, most likely, a wontfix, as 0.9.0 is now released with serviceAccount at all (so we would fix something that… is gonna go away really soon)

So there's a workaround we can apply for now (specifying `serviceAccountName`).

**Comment by a-roberts** [comment]
We've got the fix for this
```
adams-mbp:config aroberts$ k logs -f monitor-taskrun-zcdkx-pod-5d4015  --all-containers
{"level":"warn","ts":1576085694.4872673,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576085694.4882715,"logger":"fallback-logger","caller":"creds-init/main.go:40","msg":"Credentials initialized."}
{"level":"warn","ts":1576085717.5041637,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576085717.5073676,"logger":"fallback-logger","caller":"bash/main.go:64","msg":"Successfully executed command \"sh -c mkdir -p /workspace/output/pull-request\"; output "}
{"level":"warn","ts":1576085718.0339513,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576085718.0340252,"logger":"fallback-logger","caller":"pullrequest-init/main.go:44","msg":"RUNNING DOWNLOAD!"}
{"level":"info","ts":1576085718.840631,"logger":"fallback-logger","caller":"pullrequest-init/github.go:213","msg":"Writing comment 16548098 to file: /workspace/pull-request/github/comments/16548098.json"}
Setting status to pending with URL : https://github.ibm.com/api/v3/repos/AROBERTS/devops-world-example/statuses/235ac6459154c9535b2264b131c343bbb22319d7
<Response [201]>
Checking pipelinerun buildah-pipeline-run-xl8rf in namespace tekton-pipelines
Failed - pipelinerun buildah-pipeline-run-xl8rf in namespace tekton-pipelines
Set details url to: http://tekton-dashboard-tekton-pipelines.apps.hogans.os.fyre.ibm.com/#/pipelineruns
{"level":"warn","ts":1576085731.3211954,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576085731.3212638,"logger":"fallback-logger","caller":"pullrequest-init/main.go:54","msg":"RUNNING UPLOAD!"}
{"level":"info","ts":1576085731.3216715,"logger":"fallback-logger","caller":"pullrequest-init/github.go:243","msg":"Syncing path: &{Type: ID:0 Head:0xc42029c4e0 Base:0xc42029c420 Statuses:[0xc4202a0200] Comments:[0xc42002ddc0] Labels:[] Raw: RawStatus:} to pr 32"}
{"level":"info","ts":1576085731.9426217,"logger":"fallback-logger","caller":"pullrequest-init/github.go:281","msg":"Current labels: map[]"}
{"level":"info","ts":1576085731.942733,"logger":"fallback-logger","caller":"pullrequest-init/github.go:292","msg":"Creating labels [] for PR 32"}

{"level":"info","ts":1576085732.1899035,"logger":"fallback-logger","caller":"pullrequest-init/github.go:312","msg":"Setting comments for PR 32 to: [0xc42002ddc0]"}
{"level":"info","ts":1576085732.3073869,"logger":"fallback-logger","caller":"pullrequest-init/github.go:361","msg":"Deleting comment 16548098 for PR 32"}
{"level":"info","ts":1576085732.5022707,"logger":"fallback-logger","caller":"pullrequest-init/github.go:391","msg":"Creating comment ## Tekton Status Report \n\nStatus | Pipeline | PipelineRun | Namespace\n:----- | :------- | :--------------- | :--------\n[**Failed**](http://tekton-dashboard-tekton-pipelines.apps.hogans.os.fyre.ibm.com/#/namespaces/tekton-pipelines/pipelineruns/buildah-pipeline-run-xl8rf) | buildah-pipeline | buildah-pipeline-run-xl8rf | tekton-pipelines for PR 32"}
```

So specifically in the two OpenShift yamls we maintain (release and dev), for the monitor TaskRun we use serviceAccountName and we augment `tekton-triggers-minimal` (the ClusterRole) to include `list` and `get` for PipelineRuns.

**Comment by a-roberts** [comment]
Merged into 0.3.0 branch so closing, release assets to be modified next
